Created Delete Wardrobe Items and some changes to integrate the function

KhanhLeBigWow opened the pull request:
**Summary of Updates**
Back-End:
- Database schema confirmation
- API endpoint for deleting wardrobe items
- Service layer method for deletion
- Unit tests for delete functionality

Front-End:
- Created WardrobeDeleteItemForm.js
- CSS for delete button styling
- Integration with WardrobeViewComponent
- Event handling for delete confirmation
- User feedback implementation
- Manual testing of delete functionality

KhanhLeBigWow committed:
Created Delete Wardrobe Items and some changes to integrate the function

sophia-hubscher commented on front-end/src/components/WardrobeViewComponent/WardrobeViewComponent.js:
On this branch, none of my wardrobe items showed up until I created a new one. When I created the new item, everything else appeared. Not sure what that's being caused by

sophia-hubscher commented on front-end/src/components/WardrobeDeleteItemForm/WardrobeDeleteItemForm.js:
This form never appeared for me when I tried to delete an item. When I hit the trash icon, the item was removed from my wardrobe screen and I got the following console output, but I didn't get any form appearing
<img width="585" alt="Screen Shot 2024-12-07 at 12 01 39 PM" src="https://github.com/user-attachments/assets/d0eae36a-6a40-4ea4-83c8-2870e95158e6">

sophia-hubscher reviewed (changes requested):
Looks like you have a lot of great code here, but some errors are preventing it from running for me. I would try clicking through the website with your console open and investigating the errors I pointed out in my comments. Excited to see this feature when it's all the way complete—this code is looking really good!

nhan0504 commented on front-end/src/components/WardrobeDeleteItemForm/WardrobeDeleteItemForm.css:
Why is this file empty?

nhan0504 commented on front-end/src/components/WardrobeDeleteItemForm/WardrobeDeleteItemForm.js:
```diff
@@ -0,0 +1,39 @@
+    const container = document.createElement('div');
+    const deleteButton = document.createElement('button');
+    deleteButton.textContent = 'Delete Item';
+    deleteButton.classList.add('delete-item-button');
```
You're using a CSS class `delete-item-button` that is not defined anywhere because your .css file is empty

nhan0504 commented on front-end/src/components/WardrobeDeleteItemForm/WardrobeDeleteItemForm.js:
```diff
@@ -0,0 +1,39 @@
+    deleteButton.classList.add('delete-item-button');
+
+    deleteButton.addEventListener('click', async () => {
+      if (confirm('Are you sure you want to delete this item?')) {
```
This function `confirm` is not defined anywhere

nhan0504 commented on front-end/src/services/WardrobeRepositoryService.js:
```diff
@@ -158,4 +158,15 @@ export class WardrobeRepositoryService extends Service {
+
+  async deleteWardrobeItem(itemId) {
+    try {
+      await this.model.delete({ item_id: itemId }); //Adjust according to WardrobeItem's delete method
```
I don't think `model` is part of this class. How are you using it here? Also, your own comment suggests adjusting this for the `WardrobeItem` delete method, but the current code doesn’t seem to be adjusting that

nhan0504 reviewed (changes requested):
It’s not running on my end, so I recommend testing it thoroughly before pushing. If it works as intended, consider doing a short demo to ensure everything is clear and avoids unnecessary back-and-forth during review

KhanhLeBigWow commented:
Hi everyone, it's running from my end, it's strange that my css is empty... Let me take a look over them

KhanhLeBigWow committed:
Merge branch 'main' of https://github.com/nhan0504/CS326 into delete-wardrobe-items

sophia-hubscher commented on back-end/source/controller/WardrobeController.js:
```diff
@@ -145,6 +145,18 @@ class WardrobeController {
   }
+
+  async deleteWardrobeItem(req, res) {
+    const { item_id } = req.body; 
```
Make sure when you delete items, both the user IDs and item IDs match, as multiple users may have items with the same IDs

nhan0504 replied:
I don't think different user can have items with the same id as the id is autogenerated. If the id can be the same between different user this would make the get item by id endpoint not work as it will return multiple item

AshleyMcN replied:
Not sure if this was changed when we implemented backend, but when we were using indexdb the item id was the title of the item so different users could have items with the same item id. If it is still like this, the endpoint to get an item should also compare user_id

KhanhLeBigWow committed:
Merge branch 'main' of https://github.com/nhan0504/CS326 into delete-wardrobe-items

KhanhLeBigWow committed:
Merge branch 'main' of https://github.com/nhan0504/CS326 into delete-wardrobe-items

KhanhLeBigWow committed:
Merge branch 'main' of https://github.com/nhan0504/CS326 into delete-wardrobe-items

KhanhLeBigWow committed:
Merge branch 'delete-wardrobe-items' of https://github.com/nhan0504/CS326 into delete-wardrobe-items

KhanhLeBigWow commented:
> @KhanhLeBigWow I see you've been merging a lot of code into your branch here, but I'm not sure if you need a new review. Please let us know on Slack when you're ready for us to take another look 😁

Sorry for the delay. My laptop crashed when I ran another project, and it took hours to reboot and restore everything. I'll finish the review tonight asap

KhanhLeBigWow committed:
Update WardrobeDeleteItemForm.js

KhanhLeBigWow committed:
Fix some issue regarding the Delete Items Function

Ensure that the deletion logic is correctly integrated with the backend and that the UI updates appropriately after an item is deleted

KhanhLeBigWow committed:
delete deleteWardrobeItem function since I think we would not use it here

sophia-y-h commented on front-end/src/components/WardrobeViewComponent/WardrobeViewComponent.js:
```diff
@@ -313,10 +333,16 @@ export class WardrobeViewComponent extends BaseComponent {
+
       trashIcon.onclick = () => {
-        this.#wardrobeService.clearWardrobeItem(item.item_id);
+        const deleteForm = new WardrobeDeleteItemForm(item.item_id);
```
When I click the delete button, I get this error, and the delete form does not pop up. Thus, the item also isn't deleted.

<img width="587" alt="Screen Shot 2024-12-10 at 8 54 51 PM" src="https://github.com/user-attachments/assets/65abdf90-7c20-4824-9ba8-9404330203d0">

Ashley made a similar form (WardrobeAddItemForm) if you want to try to duplicate her logic for showing and hiding that. Here's a screenshot of a snippet of her code, but you'll want to search for all times it was mentioned to really get a feel for how she's doing things.

<img width="473" alt="Screen Shot 2024-12-10 at 8 54 43 PM" src="https://github.com/user-attachments/assets/7ed94e31-d423-4dde-acf9-9ab1e0a3077b">

KhanhLeBigWow committed:
Fix the delete form does not pop up

sophia-y-h commented:
The console error is resolved, but the modal isn't popping up in a visible area. You can see the delete button in the top right, which I'm assuming isn't what you wanted. Given this is due in like two hours, do you want to try the commit without the confirmation form? The form seems really cool, so if you can get that working in time, that would be awesome, but I need to head to bed pretty soon, so I would consider getting your PR out there without the confirmation form first.

<img width="814" alt="Screen Shot 2024-12-10 at 9 50 23 PM" src="https://github.com/user-attachments/assets/f2c27dbb-2ee7-4ab2-8faa-213a00522833">

KhanhLeBigWow committed:
Simplify the button

KhanhLeBigWow committed:
Update trashIcon POST request
